src/Microsoft.Sbom.Targets/GenerateSbomTask.cs

```diff
@@ -20,56 +21,62 @@ namespace Microsoft.Sbom.Targets;
-                .AddSingleton<ISourcesProvider, CGScannedExternalDocumentReferenceFileProvider>()
-                .AddSingleton<ISourcesProvider, CGScannedPackagesProvider>()
-                .AddSingleton<IAlgorithmNames, AlgorithmNames>()
-                .AddSingleton<IManifestGenerator, Generator>()
-                .AddSingleton<IMetadataProvider, LocalMetadataProvider>()
-                .AddSingleton<IMetadataProvider, SBOMApiMetadataProvider>()
-                .AddSingleton<IManifestInterface, Validator>()
-                .AddSingleton<IManifestConfigHandler, SPDX22ManifestConfigHandler>())
-            .Build();
-        this.Generator = host.Services.GetRequiredService<ISBOMGenerator>();
     }
 
     /// <inheritdoc/>
     public override bool Execute()
     {
         try
         {
+            var taskLoggingHelper = new TaskLoggingHelper(this);
+
             // Validate required args and args that take paths as input.
             if (!ValidateAndSanitizeRequiredParams() || !ValidateAndSanitizeNamespaceUriUniquePart())
             {
                 return false;
             }
 
+            var logVerbosity = ValidateAndAssignVerbosity();
+            var serilogLogVerbosity = MapSerilogLevel(logVerbosity);
+
+            var taskHost = Host.CreateDefaultBuilder()
+               .ConfigureServices((host, services) =>
+               services
+               .AddSbomTool(serilogLogVerbosity, taskLoggingHelper)
+               /* Manually adding some dependencies since `AddSbomTool()` does not add them when
+                * running the MSBuild Task from another project.
+                */
+               .AddSingleton<ISourcesProvider, SBOMPackagesProvider>()
+               .AddSingleton<ISourcesProvider, CGExternalDocumentReferenceProvider>()
+               .AddSingleton<ISourcesProvider, DirectoryTraversingFileToJsonProvider>()
+               .AddSingleton<ISourcesProvider, ExternalDocumentReferenceFileProvider>()
+               .AddSingleton<ISourcesProvider, ExternalDocumentReferenceProvider>()
+               .AddSingleton<ISourcesProvider, FileListBasedFileToJsonProvider>()
+               .AddSingleton<ISourcesProvider, SbomFileBasedFileToJsonProvider>()
+               .AddSingleton<ISourcesProvider, CGScannedExternalDocumentReferenceFileProvider>()
+               .AddSingleton<ISourcesProvider, CGScannedPackagesProvider>()
+               .AddSingleton<IAlgorithmNames, AlgorithmNames>()
+               .AddSingleton<IManifestGenerator, Generator>()
+               .AddSingleton<IMetadataProvider, LocalMetadataProvider>()
+               .AddSingleton<IMetadataProvider, SBOMApiMetadataProvider>()
+               .AddSingleton<IManifestInterface, Validator>()
+               .AddSingleton<IManifestConfigHandler, SPDX22ManifestConfigHandler>())
+               .Build();
+
+            var generator = taskHost.Services.GetRequiredService<ISBOMGenerator>();
```

Comment:
Big change here. Instead of putting the DI container in the constructor, I moved it into `Execute` so that we could call the Verbosity-defaulting assignment, so we could flow that defaulted verbosity into the Serilog configuration calls.